### PR TITLE
Allow File Provider to use representations endpoints for download to …

### PR DESCRIPTION
**Pull request by jdizoglio**
…retrieve assets with the addition of background tasks awareness.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

**Comment by jdizoglio** [comment]
retry

**Comment by jdizoglio** [comment]
test this please

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h [hunk 1]
```diff
@@ -12,17 +12,59 @@
...
+ @return instance of BOXFileRepresentationDownloadRequest
+ */
 - (instancetype)initWithLocalDestination:(NSString *)destinationPath
                                   fileID:(NSString *)fileID
                           representation:(BOXRepresentation *)representation;
 
+/**
+ Similar to the above init method, request will download file into destinationPath,
+ and the file download can continue running in the background even if app is not running
+ If downloadTask is provided, it will be used to execute/reconnect with the existing download task
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h]: if associateID is provided,...

Could we update equivalent comment for BOXFileDownloadRequest as well?

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h [hunk 2]
```diff
@@ -12,17 +12,59 @@
...
+ downloadTaskReplacedBlock is to notify when the downloadTask used by the request has been changed,
+ useful to reconnecting to ongoing download task upon app restart
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h]: I think this comment is outdated "downloadTaskReplacedBlock...app restart"

could we remove for BOXFileDownloadRequest as well?

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h (hunk 1 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h (hunk 2 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.h]: done

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+File.h [hunk 3]
```diff
@@ -362,6 +362,21 @@
                                                                   toLocalFilePath:(NSString *)localFilePath
                                                                    representation:(BOXRepresentation *)representation;
 
+/**
+ *  Generate a request to download a given representation of a file to a local filepath.
+ *
+ *  @param fileID          File ID.
+ *  @param localFilePath   Local filepath.
+ *  @param representation  BOXRepresentation to be downloaded
+ *  @param associateID A unique ID used to recover background tasks. Providing an ID associated with a running background
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+File.h:371]: @param associateID   A unique ID to associate with a background download task. Required together with localFilePath to trigger a background download.

**Review by thuyBox — APPROVED** [review]
after updating last comment

**Pull request merged**